Tasks/UnityBuild/UnityBuildV3/unity-versioning.ts

```diff
@@ -0,0 +1,306 @@
+            buildNumber
+          );
+        } else if (buildPlatform === BuildPlatform.VisionOS) {
+          buildCode = UnityVersioningTools.setBuildNumber(projectPath, {
+            VisionOS: buildNumber,
+          }).VisionOS;
+        } else if (buildPlatform === BuildPlatform.IOS) {
+          buildCode = UnityVersioningTools.setBuildNumber(projectPath, {
+            iPhone: buildNumber,
+          }).iPhone;
+        } else if (buildPlatform === BuildPlatform.TVOS) {
+          buildCode = UnityVersioningTools.setBuildNumber(projectPath, {
+            tvOS: buildNumber,
+          }).tvOS;
+        } else {
+          throw new Error(`Invalid build platform: ${buildPlatform}`);
+        }
+      } else {
+        throw new Error(`Invalid build number mode: ${buildNumberMode}`);
+      }
+    } else {
+      // By passing 0 increments, we'll essentially just read the current version.
+      if (
+        buildPlatform !== BuildPlatform.VisionOS &&
+        buildPlatform !== BuildPlatform.IOS &&
+        buildPlatform !== BuildPlatform.TVOS &&
+        buildPlatform !== BuildPlatform.Android
+      ) {
+        buildCode = UnityVersioningTools.incrementBuildNumber(
+          projectPath,
+          {}
+        ).Standalone;
+      } else if (buildPlatform === BuildPlatform.Android) {
+        buildCode = UnityVersioningTools.incrementAndroidBundleVersionCode(
+          projectPath,
+          0
+        );
+      } else if (buildPlatform === BuildPlatform.VisionOS) {
+        buildCode = UnityVersioningTools.incrementBuildNumber(
+          projectPath,
+          {}
+        ).VisionOS;
+      } else if (buildPlatform === BuildPlatform.IOS) {
+        buildCode = UnityVersioningTools.incrementBuildNumber(
+          projectPath,
+          {}
+        ).iPhone;
+      } else if (buildPlatform === BuildPlatform.TVOS) {
+        buildCode = UnityVersioningTools.incrementBuildNumber(
+          projectPath,
+          {}
+        ).tvOS;
+      } else {
+        throw new Error(`Invalid build platform: ${buildPlatform}`);
+      }
+    }
+
+    const commitChanges = tl.getBoolInput(
+      projectVersioningCommitChangesVariableName,
+      true
+    )!;
+
+    let gitTag: string | "" = "";
+
+    // Does the user want to commit changes to the repositoryß
+    if (commitChanges) {
+      const commitChangesUserName = tl.getInput(
+        projectVersioningCommitChangesUserNameVariableName,
+        true
+      )!;
+      const commitChangesUserMail = tl.getInput(
+        projectVersioningCommitChangesUserMailVariableName,
+        true
+      )!;
+
+      const createTag = tl.getBoolInput(
+        projectVersioningCreateTagVariableName,
+        true
+      )!;
+
+      tl.execSync("git", ["config", "user.name", commitChangesUserName]);
+      tl.execSync("git", ["config", "user.email", commitChangesUserMail]);
+      tl.execSync("git", ["add", "."]);
+      tl.execSync("git", ["commit", "-m Azure Pipelines Build"]);
```

Comment:
The message would be nice if it could also be configured even if it had a default value because in my case I like to: https://www.conventionalcommits.org/en/v1.0.0/

Reply:
It will be configurable for sure as this feature matures towards release.